server/src/main/java/org/elasticsearch/action/search/TransportOpenPointInTimeAction.java

```diff
@@ -125,6 +132,63 @@ public SearchPhase newSearchPhase(
             boolean preFilter,
             ThreadPool threadPool,
             SearchResponse.Clusters clusters
+        ) {
+            if (SearchService.canRewriteToMatchNone(searchRequest.source())) {
+                return new CanMatchPreFilterSearchPhase(
```

Comment:
> why do we return a CanMatchPreFilterSearchPhase here if there is no filter (or a match-all) ?

That should not be what's happening. The intention is to run the can match phase only when it makes sense (when there is an index filter which may rewrite to match_none), otherwise skip the can match phase entirely (else branch). 

If there is not index_filter, or if the index_filter is a match_all query (it is impossible for it to hold suggest or aggs), the canRewriteToMatchNone will return false, hence we will skip the can match phase. Does this make sense?

Reply:
Yeah makes sense, thanks @javanna !